### PR TITLE
CE-2902 added full template classification labels

### PR DESCRIPTION
**Pull request by lgarczewski**
@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/TemplateClassification/TemplateClassification.i18n.php [hunk 1]
```diff
@@ -4,12 +4,44 @@
...
+        'template-classification-description-design' => 'This template is used to design or structure article pages.',
+
+        'template-classification-type-navigation' => 'Navigation',
+        'template-classification-description-navigation' => 'This template is used for navigation.',
+
+        'template-classification-type-nonarticle' => 'Non-article',
+        'template-classification-description-nonarticle' => 'This template is not used on article pages.',
+
+        'template-classification-type-dontknow' => 'Unknown',
+        'template-classification-description-dontknow' => "I don't know.",
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassification.i18n.php]: All these new messages need QQQ docs.

File: extensions/wikia/TemplateClassification/TemplateClassificationController.class.php [hunk 2]
```diff
@@ -23,6 +23,7 @@ public function getTemplateClassificationEditForm() {
 				 * template-classification-type-unclassified
 				 */
 				'name' => wfMessage( "template-classification-type-{$type}" )->escaped(),
+				'description' => wfMessage( "template-classification-description-{$type}" )->escaped(),
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassificationController.class.php]: Since the message key is being constructed with a variable, similar to the message use above, this should have a comment above it with the possible message strings that can be used here so it is possible to find where they are used with grep.

File: extensions/wikia/TemplateClassification/TemplateClassificationView.php [hunk 3]
```diff
@@ -13,14 +13,17 @@ class View {
...
+		$templateType = lcfirst( ( new \TemplateClassificationMockService() )->getTemplateType( $articleId ) );
 		/**
 		 * template-classification-type-unclassified
 		 * template-classification-type-infobox
 		 * template-classification-type-navbox
 		 * template-classification-type-quote
 		 */
-		$templateName = wfMessage( "template-classification-type-{$templateType}" )->escaped();
+		$templateName = wfMessage(
+			"template-classification-indicator",
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassificationView.php]: Should use single quotes per conventions since it isn't using string interpolation.

File: extensions/wikia/TemplateClassification/TemplateClassificationView.php [hunk 4]
```diff
@@ -13,14 +13,17 @@ class View {
...
+			wfMessage( "template-classification-type-{$templateType}" )
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassificationView.php:25]: Since this is being escaped as a parameter to the `template-classification-indicator` message, this should have an output method of `->plain()` or `->text()`. If you don't add an explicit output type it will use `->parse()` which means it would be double escaped and is an unnecessary call to the parser.

File: includes/wikia/TemplateClassification.class.php [hunk 5]
```diff
@@ -30,6 +38,15 @@ class TemplateClassification {
 		self::TEMPLATE_QUOTE,
 		self::TEMPLATE_NAVBOX,
 		self::TEMPLATE_UNCLASSIFIED,
```
Comment (kamilkoterba) [inline includes/wikia/TemplateClassification.class.php:40]: TEMPLATE_UNCLASSIFIED was about to be replaced with TEMPLATE_UNKNOWN

File: includes/wikia/TemplateClassification.class.php [hunk 6]
```diff
@@ -30,6 +38,15 @@ class TemplateClassification {
...
+		self::TEMPLATE_NAVBOX,
+		self::TEMPLATE_FLAG,
+		self::TEMPLATE_REF,
+		self::TEMPLATE_MEDIA,
+		self::TEMPLATE_DATA,
+		self::TEMPLATE_DESIGN,
+		self::TEMPLATE_NAV,
+		self::TEMPLATE_NOT_ART,
+		self::TEMPLATE_DONTKNOW,
```
Comment (kamilkoterba) [inline includes/wikia/TemplateClassification.class.php]: DONTKNOW is just user facing message fallback on modal for TEMPLATE_UNKNOWN type so shouldn't be visible here

File: skins/oasis/modules/TemplateDraftModuleController.class.php [hunk 7]
```diff
@@ -11,7 +11,8 @@ public function executeCreate( $params ) {
 			AssetsManager::getInstance()->getSassCommonURL( "skins/oasis/css/modules/TemplateDraftModule.scss" )
 		);
 
-		$subpage = wfMessage( 'templatedraft-subpage' )->inContentLanguage()->escaped();
+		$subpage = 'foo';
+//		$subpage = wfMessage( 'templatedraft-subpage' )->inContentLanguage()->escaped();
```
Comment (Grunny) [inline skins/oasis/modules/TemplateDraftModuleController.class.php]: Testing code left over?

File: includes/wikia/TemplateClassification.class.php [hunk 8]
```diff
@@ -52,8 +68,7 @@ public function getType() {
 
 	/**
 	 * Checks if the template is of the given type.
-	 * @param $type One of values from the $templateTypes array
-	 * @return bool
+
```
Comment (Grunny) [inline includes/wikia/TemplateClassification.class.php]: Why was this removed?

File: includes/wikia/TemplateClassification.class.php (hunk 5 above)
Comment (lgarczewski) [inline includes/wikia/TemplateClassification.class.php:40]: Per discussion: we need to keep UNCLASSIFIED for now because it has a slightly different meaning. Once we hook up the Template Conversion flow to the service UNCLASSIFIED will be removed.

File: extensions/wikia/TemplateClassification/TemplateClassification.i18n.php [hunk 9]
```diff
@@ -4,12 +4,44 @@
...
+        'template-classification-description-design' => 'This template is used to design or structure article pages.',
+
+        'template-classification-type-navigation' => 'Navigation',
+        'template-classification-description-navigation' => 'This template is used for navigation.',
+
+        'template-classification-type-nonarticle' => 'Non-article',
+        'template-classification-description-nonarticle' => 'This template is not used on article pages.',
+
+        'template-classification-type-unknown' => 'Unknown',
+        'template-classification-description-unknown' => "I don't know.",
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassification.i18n.php]: Some whitespace issues, the messages above are all indented with spaces instead of tabs.

File: extensions/wikia/TemplateClassification/TemplateClassification.i18n.php [hunk 10]
```diff
@@ -17,10 +49,33 @@
...
+        'template-classification-description-quote' => 'Description (additional label) for the quote type. This is shown in dialog box under the type name to explain what each type is.',
+        'template-classification-description-data' => 'Description (additional label) for the data type. This is shown in dialog box under the type name to explain what each type is.',
+        'template-classification-description-design' => 'Description (additional label) for the design type. This is shown in dialog box under the type name to explain what each type is.',
+        'template-classification-description-navigation' => 'Description (additional label) for the navigation type. This is shown in dialog box under the type name to explain what each type is.',
+        'template-classification-description-nonarticle' => 'Description (additional label) for the non-article type. This is shown in dialog box under the type name to explain what each type is.',
+        'template-classification-description-unknown' => 'Description (additional label) for the unknown template type. This is shown in dialog box under the type name to explain what each type is.',
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/TemplateClassification.i18n.php]: Same here.

**Comment by Grunny** [comment]
Other than last whitespace comments, LGTM. So, ready to go if all tested. :)

**Comment by kamilkoterba** [comment]
LGTM if tested
